pkgs/applications/office/sleek/default.nix

```diff
@@ -0,0 +1,71 @@
+{ mkYarnPackage
+, fetchFromGitHub
+, electron_18
+, wrapGAppsHook
+, lib
+}:
+
+mkYarnPackage rec {
```

Comment:
I think mkYarnPackage is missing its package.json.

Reply:
But the package.json is in the src, it's even being patched successfully. What do you mean it's missing?

Reply:
I think that is triggering IFD which is forbidden in nixpkgs

Reply:
I'm missing context. Why is it triggering IFD?

Reply:
mkYarnPackage falls internally back to IFD. We should probably disable that for nixpkgs. Please see other packages how to provide the package.json.

Reply:
Did I do it correctly in the latest commit?

Reply:
ofborg eval still fails with the following error which means IFD is used

```
nix-env failed:
error:
       … while evaluating call site

       at «none»:0: (source not available)

       … while calling anonymous lambda

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/.gc-of-borg-outpaths.nix:47:12:

           46|   tweak = lib.mapAttrs
           47|     (name: val:
             |            ^
           48|       if name == "recurseForDerivations" then true

       … while evaluating call site

       at «none»:0: (source not available)

       … while calling 'g'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/lib/attrsets.nix:595:19:

          594|           g =
          595|             name: value:
             |                   ^
          596|             if isAttrs value && cond value

       … while evaluating call site

       at «none»:0: (source not available)

       … while calling anonymous lambda

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/top-level/release-lib.nix:148:38:

          147|      set of meta.platforms values. */
          148|   packagePlatforms = mapAttrs (name: value:
             |                                      ^
          149|       if isDerivation value then

       … while evaluating the attribute 'meta.hydraPlatforms'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/stdenv/generic/make-derivation.nix:558:13:

          557|      inherit passthru overrideAttrs;
          558|      inherit meta;
             |             ^
          559|    } //

       … while evaluating call site

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/stdenv/generic/make-derivation.nix:505:10:

          504|
          505|   meta = checkMeta.commonMeta { inherit validity attrs pos references; };
             |          ^
          506|   validity = checkMeta.assertValidity { inherit meta attrs; };

       … while calling 'commonMeta'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/stdenv/generic/check-meta.nix:401:16:

          400|   #   validity = checkMeta.assertValidity { inherit meta attrs; };
          401|   commonMeta = { validity, attrs, pos ? null, references ? [ ] }:
             |                ^
          402|     let

       … while evaluating the attribute 'meta'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:391:7:

          390|
          391|       meta = {
             |       ^
          392|         inherit (nodejs.meta) platforms;

       … while evaluating call site

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:393:12:

          392|         inherit (nodejs.meta) platforms;
          393|       } // lib.optionalAttrs (package ? description) { inherit (package) description; }
             |            ^
          394|         // lib.optionalAttrs (package ? homepage) { inherit (package) homepage; }

       … while calling 'optionalAttrs'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/lib/attrsets.nix:675:5:

          674|     # The attribute set to return if `cond` is `true`.
          675|     as:
             |     ^
          676|     if cond then as else {};

       … while evaluating call site

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:269:17:

          268|     let
          269|       package = lib.importJSON packageJSON;
             |                 ^
          270|       pname = package.name;

       … while calling 'importJSON'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/lib/trivial.nix:312:16:

          311|   */
          312|   importJSON = path:
             |                ^
          313|     builtins.fromJSON (builtins.readFile path);

       … while realising the context of path '/nix/store/mbhbjyk01mlpakmwlzh673hkyi6lhxar-source/package.json'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/lib/trivial.nix:313:24:

          312|   importJSON = path:
          313|     builtins.fromJSON (builtins.readFile path);
             |                        ^
          314|

       error: path '/nix/store/nj2j3nn03j2p5zwv7yb8d29rihhkhdjq-source.drv' is not valid

```

Reply:
oh, IFD is used before it's converting the JSON to nix and then IFDing that. I don't know how else to specify the package.json though. A lot of other node programs do that.

Reply:
the entire package.json would have to be converted to nix and then inlined into the default.nix

Reply:
/me is looking at more other yarn packages